src/mem.rs

```diff
@@ -0,0 +1,60 @@
+// NOTE Copied verbatim from the rlibc crate
```

Comment:
@shepmaster 

Given that the ultimate goal is to have this crate as part of rust-lang/rust releases. It makes sense to me to also provide these `mem*` functions so users don't have to reach out for an external (crates.io) crate.